js/src/components/app/StateSharing.tsx

```diff
@@ -36,33 +42,58 @@ export function TopLevelShare() {
   if (!authed) {
     return (
       <Flex flex="1" alignItems="center">
-        <Popover trigger="hover" placement="bottom-start">
-          <PopoverTrigger>
-            <Button
-              rightIcon={<ExternalLinkIcon />}
-              size="sm"
-              onClick={() => {
-                trackShareState({ name: "enable" });
-                window.open(
-                  `${PUBLIC_CLOUD_WEB_URL}/settings?utm_source=recce_oss&utm_content=enable_sharing_button#tokens`,
-                  "_blank",
-                );
-              }}>
-              Enable sharing
-            </Button>
-          </PopoverTrigger>
-          <PopoverContent bg="black" color="white" sx={{ width: "max-content" }}>
-            <PopoverBody fontSize="sm">
-              API token required.{" "}
-              <Link
-                href="https://docs.datarecce.io/recce-cloud/share-recce-session-securely"
-                target="_blank"
-                textDecoration="underline">
-                Learn more
-              </Link>
-              .
-            </PopoverBody>
-          </PopoverContent>
+        <Popover placement="bottom-start" initialFocusRef={initialFocusRef}>
+          {({ onClose }) => (
+            <>
+              <PopoverTrigger>
+                <Button size="sm" variant="outline" leftIcon={<TbCloudUpload />}>
+                  Share
+                </Button>
+              </PopoverTrigger>
+              <PopoverContent bg="gray.200" sx={{ width: "max-content" }}>
+                <PopoverHeader fontWeight="semibold">Enable sharing</PopoverHeader>
+                <PopoverCloseButton />
+                <PopoverBody fontSize="sm">
+                  API token required.{" "}
+                  <Link
+                    href="https://docs.datarecce.io/recce-cloud/share-recce-session-securely"
+                    target="_blank"
+                    textDecoration="underline">
+                    Learn more
+                  </Link>
+                  .
+                </PopoverBody>
+                <PopoverFooter
+                  border="0"
+                  display="flex"
+                  alignItems="center"
+                  justifyContent="flex-end"
+                  pb={4}>
+                  <ButtonGroup size="sm">
+                    <Link
+                      ref={initialFocusRef}
+                      onClick={() => {
+                        trackShareState({ name: "enable" });
+                        onClose();
+                      }}
+                      href={`${PUBLIC_CLOUD_WEB_URL}/signin?callbackUrl=/relaunch&utm_source=recce_oss&utm_content=enable_sharing_button#tokens`}
```

Comment:
Is it better to directly link `${PUBLIC_CLOUD_WEB_URL}/relaunch?utm_source=...`?
If the user is already authenticated to Recce Cloud, it will go to the relaunch page directly. The user doesn't need to log in again.